SimMuon/Configuration/python/customizeMuonDigi.py

```diff
@@ -0,0 +1,353 @@
+    process = customize_random_GEMDigi(process)
+    process = customize_random_ME0Digi(process)
+    process = customize_mix_addGEM(process)
+    process = customize_mix_addME0(process)
+    process.doAllDigi = cms.Sequence(
+        process.calDigi + 
+        process.muonDigi +
+        process.simMuonGEMDigi +
+        process.simMuonME0Digi
+    )
+    process.pdigi = cms.Sequence(
+        cms.SequencePlaceholder("randomEngineStateProducer")*
+        cms.SequencePlaceholder("mix")*
+        process.doAllDigi*
+        process.addPileupInfo
+    )
+    process = append_GEMDigi_event(process)
+    return process
+
+
+# insert the GEMDigi and GEMPadDigi collection to the event
+def append_GEMDigi_event(process):
+    alist=['AODSIM','RECOSIM','FEVTSIM','FEVTDEBUG','FEVTDEBUGHLT','RECODEBUG','RAWRECOSIMHLT','RAWRECODEBUGHLT']
+    for a in alist:
+        b=a+'output'
+        if hasattr(process,b):
+            getattr(process,b).outputCommands.append('keep *_g4SimHits_Muon*_*')
+            getattr(process,b).outputCommands.append('keep *_*Muon*_*_*')
+
+    return process
+
+# Customizations for the background
+def customize_digi_noGEMbkg(process):
+    process.simMuonGEMDigis.doBkgNoise = False
+    return process
+
+def customize_digi_noGEMsafety(process):
+    process.simMuonGEMDigis.rateFact = 1
+    return process
+
+# Customizations for the background (to be updated once the realistic digi is in place)
+def customize_digi_noME0bkg(process):
+    process.simMuonME0PseudoDigis.simulateElectronBkg = False
+    process.simMuonME0PseudoDigis.simulateNeutralBkg = False
+    return process
+
+def customize_digi_noME0safety(process):
+    process.simMuonME0PseudoDigis.rateFact = 1
+    return process
+
+# Customizations for the background
+def customize_digi_noRPCbkg(process):
+    process.simMuonRPCDigis.doBkgNoise = False
+    return process
+
+# adding re-digi costumisation - to be used for dedicated trigger studies
+def customise_rpcRedigi(process):
+    process.load('Configuration.StandardSequences.Digi_cff')
+    process.simMuonRPCReDigis = process.simMuonRPCDigis.clone()
+    process.simMuonRPCReDigis.digiRPCModelConfig = cms.PSet(
+        Frate = cms.double(1.0),
+        Gate = cms.double(1.0),
+        IRPC_electronics_jitter = cms.double(0.1),
+        IRPC_time_resolution = cms.double(1.0),
+        Nbxing = cms.int32(799),
+        BX_range = cms.int32(400),
+        Rate = cms.double(0.0),
+        averageClusterSize = cms.double(1.5),
+        averageEfficiency = cms.double(0.95),
+        cosmics = cms.bool(False),
+        deltatimeAdjacentStrip = cms.double(3.0),
+        linkGateWidth = cms.double(1.0),
+        printOutDigitizer = cms.bool(False),
+        signalPropagationSpeed = cms.double(0.66),
+        timeJitter = cms.double(1.0),
+        timeResolution = cms.double(2.5),
+        timingRPCOffset = cms.double(50.0)
+    ),
+    process.simMuonRPCReDigis.digiIRPCModelConfig = cms.PSet(
```

Comment:
@bapavlov @jhgoh can you have a look on the RPC resolution constants here, please.